components/renku_data_services/platform/orm.py

```diff
@@ -50,3 +52,42 @@ def dump(self) -> models.PlatformConfig:
             creation_date=self.creation_date,
             updated_at=self.updated_at,
         )
+
+
+class UrlRedirectsORM(BaseORM):
+    """The url redirects."""
+
+    __tablename__ = "url_redirects"
+
+    id: Mapped[ULID] = mapped_column("id", ULIDType, primary_key=True, default_factory=lambda: str(ULID()), init=False)
+
+    source_url: Mapped[str] = mapped_column("source_url", String(), default="", unique=True, index=True)
+    """The source URL for the redirect."""
+
+    target_url: Mapped[str] = mapped_column("target_url", String(), default="", index=True)
+    """The target URL for the redirect."""
+
```

Comment:
Not sure what is the point with the defaults. Both of these have to be provided and they cannot be "" or NULL.


```suggestion
    source_url: Mapped[str] = mapped_column("source_url", String(), unique=True, index=True)
    """The source URL for the redirect."""

    target_url: Mapped[str] = mapped_column("target_url", String(), index=True)
    """The target URL for the redirect."""

```